src/zebras/_freq.py

```diff
@@ -126,21 +136,39 @@ def __get_validators__(cls):
         yield freq
 
     @classmethod
-    def from_pandas(cls, freq) -> Freq:
-        if not isinstance(freq, str):
-            if hasattr(freq, "freqstr"):
-                freq = freq.freqstr
-            else:
-                raise ValueError(f"Invalid freq {freq}: {type(freq)}")
+    def get_weekday_offset(cls, freqstr):
+        if freqstr.upper() == "W":
+            return 0
+
+        assert (
+            len(freqstr) == 5
+            and freqstr[:2] == "W-"
+            and freqstr[2:] in list(day_offsets.keys())
```

Comment:
```suggestion
            and freqstr[2:] in day_offsets
```

Note: `list` would also make this `O(n)` instead of `O(1)`.